protocol/contracts/libraries/Minting/LibWellMinting.sol

```diff
@@ -172,49 +214,29 @@ library LibWellMinting {
             uint40(s.season.timestamp),
             pumps[0].data
         ) returns (uint[] memory twaReserves, bytes memory snapshot) {
-            IERC20[] memory tokens = IWell(well).tokens();
-            (
-                uint256[] memory ratios,
-                uint256 beanIndex,
-                bool success
-            ) = LibWell.getRatiosAndBeanIndex(tokens, block.timestamp.sub(s.season.timestamp));
-
-            // If the Bean reserve is less than the minimum, the minting oracle should be considered off.
-            if (twaReserves[beanIndex] < C.WELL_MINIMUM_BEAN_BALANCE) {
-                return (0, snapshot, new uint256[](0), new uint256[](0));
-            }
-
-            // If the USD Oracle oracle call fails, the minting oracle should be considered off.
-            if (!success) {
-                return (0, snapshot, twaReserves, new uint256[](0));
-            }
-
-            Call memory wellFunction = IWell(well).wellFunction();
-            // Delta B is the difference between the target Bean reserve at the peg price
-            // and the time weighted average Bean balance in the Well.
-            int256 deltaB = int256(IBeanstalkWellFunction(wellFunction.target).calcReserveAtRatioSwap(
-                twaReserves,
-                beanIndex,
-                ratios,
-                wellFunction.data
-            )).sub(int256(twaReserves[beanIndex]));
-
-            return (deltaB, snapshot, twaReserves, ratios);
+                                // well, reserves, snapshot  lookback
+            return (getDeltaBInfo(well, twaReserves, snapshot, block.timestamp.sub(s.season.timestamp)));
         }
         catch {
             // if the pump fails, return all 0s to avoid the sunrise reverting.
             return (0, new bytes(0), new uint256[](0), new uint256[](0));
         }
     }
 
-    // Remove in next BIP.
-    function checkShouldTurnOnMinting(address well) internal view returns (bool) {
+    /**
+     * @dev Calculates the instantaneous delta B for a given Well address.
+     * @param well The address of the Well.
+     * @return deltaB The instantaneous delta B balance since the last `capture` call.
+     */
+    function instantaneousDeltaB(address well) internal view returns (int256) {
+
         AppStorage storage s = LibAppStorage.diamondStorage();
-        if (well == C.BEAN_ETH_WELL) {
-            if (s.season.current < s.season.beanEthStartMintingSeason) {
-                return false;
-            }
-        }
-        return true;
+        Call[] memory pumps = IWell(well).pumps();
+                                                                                            // well address , data[]
+        uint[] memory instReserves = IInstantaneousPump(pumps[0].target).readInstantaneousReserves(well, C.BYTES_ZERO);
```

Comment:
Hmm, you are right, it provides another layer of defense on pump errors. Probably didnt include it since i mistakenly thought pump failure was also handled in  ```getDeltaBInfo```. Added here [e8f74f0](https://github.com/BeanstalkFarms/Beanstalk/pull/802/commits/e8f74f0b52d65369bc0d47a4dee4b2a621c5bb0c)

Reply:
Removed the duplicate declaration here [092a4a4](https://github.com/BeanstalkFarms/Beanstalk/pull/802/commits/092a4a4188fafd1bafd4ba92c38c4d641afb813f)